test/View/Helper/FormDateTimeSelectTest.php

```diff
@@ -299,4 +300,23 @@ public function testRendersDatesWithEsCLLocaleDatePattern(): void
             '<html>' . $this->helper->render($element) . '</html>'
         );
     }
+
+    public function testRendersDatesOnlyWithNumericValuesInSelectOptions(): void
+    {
+        $element = new DateTimeSelect('foo');
+        $element->setMinYear(2023);
+        $element->setMaxYear(2023);
+
+        $availableLocales = IntlCalendar::getAvailableLocales();
```

Comment:
I'm skeptical about non-deterministic tests: this array could be empty in a computer, or have 785 locales in another one.

I'd prefer to pick, let's say, 3 known locales useful for our purpose, assert they are in `IntlCalendar::getAvailableLocales()`, and then cycle on those 3

Reply:
In my computer I get the following locales (generic, no country part) with non numeric formattng:
```
ar (Arabic) => ٠٥ ٢٧ ٢٢ ٣٣ ٤٤
as (Assamese) => ০৫ ২৭ ২২ ৩৩ ৪৪
bn (Bangla) => ০৫ ২৭ ২২ ৩৩ ৪৪
ccp (Chakma) => 𑄶𑄻 𑄸𑄽 𑄸𑄸 𑄹𑄹 𑄺𑄺
ckb (Central Kurdish) => ٠٥ ٢٧ ٢٢ ٣٣ ٤٤
dz (Dzongkha) => ༠༥ ༢༧ ༢༢ ༣༣ ༤༤
fa (Persian) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
ks (Kashmiri) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
lrc (Northern Luri) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
mni (Manipuri) => ০৫ ২৭ ২২ ৩৩ ৪৪
mr (Marathi) => ०५ २७ २२ ३३ ४४
my (Burmese) => ၀၅ ၂၇ ၂၂ ၃၃ ၄၄
mzn (Mazanderani) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
ne (Nepali) => ०५ २७ २२ ३३ ४४
ps (Pashto) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
sa (Sanskrit) => ०५ २७ २२ ३३ ४४
sat (Santali) => ᱐᱕ ᱒᱗ ᱒᱒ ᱓᱓ ᱔᱔
sd (Sindhi) => ٠٥ ٢٧ ٢٢ ٣٣ ٤٤

```

We could use all of them (without the different variants) or we could pick, like you suggested, three of them with and/or without country specs:

```
ar (Arabic) => ٠٥ ٢٧ ٢٢ ٣٣ ٤٤
fa (Persian) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
sa (Sanskrit) => ०५ २७ २२ ३३ ४४
```
or
```
ar_JO (Arabic (Jordan)) => ٠٥ ٢٧ ٢٢ ٣٣ ٤٤
fa_IR (Persian (Iran)) => ۰۵ ۲۷ ۲۲ ۳۳ ۴۴
sa_IN (Sanskrit (India)) => ०५ २७ २२ ३३ ४४
```

or both

Reply:
...and what about just skipping the test if a test-locale is not available?

```php
    public function testRendersDatesOnlyWithNumericValuesInSelectOptions(): void
    {
        $element = new DateTimeSelect('foo');
        $element->setMinYear(2023);
        $element->setMaxYear(2023);

        $availableLocales = IntlCalendar::getAvailableLocales();
        $testLocales      = ['ar', 'fa', 'sa'];

        foreach ($testLocales as $locale) {
            if (! in_array($locale, $availableLocales, true)) {
                self::markTestSkipped("Locale '{$locale}' is not available");
            }
            $this->helper->setLocale($locale);
            $this->helper->setDateType(IntlDateFormatter::SHORT);
            self::assertDoesNotMatchRegularExpression(
                '/<option value="[^"]*[^0-9"]+[^"]*">/',
                $this->helper->render($element),
                "Option value with locale={$locale} contains non numeric characters!"
            );
        }
    }
```

Reply:
Mmm, ok: let's keep the `foreach` over `$availableLocales = IntlCalendar::getAvailableLocales();`, but let's assert first that that array contains both `en` and `ar`

Reply:
A clarification about my previous comment: when I listed the no-country troublesome locales, I meant that I get a lot of different variants (~80-100), but skipping the ones with country spec we are left with ~20 entries (the ones I listed), not that I only get those locales without the country part (I previously listed all the troublesome locales in the related issue)

Q: why do we need asserting the presence of a locale instead of just skipping the test for that locale?

What we want is to to make sure that if a locale is available and it's being used with "laminas-form datetime selects" it does not generate invalid option values. 
If a locale is not available in a php installation, then the "intl-formatter" will just fall back to the default locale or to `en_US_POSIX`. 
The availability of a locale (troublesome or not for our case) is not the subject of this test. 
Or am I just not getting what you really mean with "assert"? :-)

kind regards

Reply:
This test must cover the `ar` case. If we do not `self::assertContains('ar', $availableLocales);`, then maybe your computer doesn`t have `ar`, Github Actions doesn't have it either, the test pass but we are not covering the case.

Reply:
right!